tests/test_squeues.py

```diff
@@ -114,7 +114,7 @@ def test_nonserializable_object(self):
         # Trigger Twisted bug #7989
         import twisted.persisted.styles  # NOQA
         q = self.queue()
-        self.assertRaises(ValueError, q.push, lambda x: x)
+        self.assertRaises((ValueError, AttributeError), q.push, lambda x: x)
```

Comment:
right @kmike , [only `ValueError` is caught: ](https://github.com/scrapy/scrapy/blob/master/scrapy/core/scheduler.py#L84) to return `True` or `False` to decide where to put it. So this test fixing is definitely not right.